passes/pmgen/peepopt_shiftadd.pmg

```diff
@@ -103,8 +103,16 @@ code
 		new_a.append(old_a);
 	} else {
 		// data >> (...+c) transformed to data[MAX:c] >> (...)
-		new_a.append(old_a.extract_end(offset));
-
+		if(offset < GetSize(old_a)) // some signal bits left?
+			new_a.append(old_a.extract_end(offset));
+		else {
+			if(shift->type.in($shiftx))
+				log_warning("at %s: result of indexed part-selection is always constant (selecting from '%s' with index '%s + %d')\n", \
```

Comment:
`shift->get_src_attribute()` is not guaranteed not to be empty, please add minimal handling of this case. At least for example
```
    std::string location = shift->get_src_attribute();
    if (location.empty())
        location = shift->name;
```
then use `location` in the phrasing. I think I saw other places in Yosys handle this the same way.

Reply:
Handled this case, the warning is not the cleanest if the src-attribute is missing but I think it is acceptable, especially because without it a 'good' warning is hard to do anyway.